.github/workflows/test.yml

```diff
@@ -71,6 +94,7 @@ jobs:
 
   e-commerce:
     runs-on: ubuntu-latest
+    needs: [typecheck]
```

Comment:
Now depends on the `typecheck` job. The `build` step also run a typecheck on the e-commerce demo